roles/openshift_health_checker/openshift_checks/fluentd_config.py

```diff
@@ -0,0 +1,188 @@
+"""
+Module for performing checks on a Fluentd logging deployment configuration
+"""
+
+import os
+import json
+
+from openshift_checks import OpenShiftCheck, get_var, OpenShiftCheckException
+
+
+class FluentdConfig(OpenShiftCheck):
+    """Module that checks logging configuration of an integrated logging Fluentd deployment"""
+    name = "fluentd_config"
+    tags = ["health"]
+
+    logging_namespace = "logging"
+
+    @classmethod
+    def is_active(cls, task_vars):
+        return super(FluentdConfig, cls).is_active(task_vars)
+
+    def run(self, tmp, task_vars):
+        """Check that Fluentd has running pods, and that its logging config matches Docker's logging config."""
+        self.logging_namespace = get_var(task_vars, "openshift_logging_namespace", default=self.logging_namespace)
+        config_error = self.check_logging_config(task_vars)
+        if config_error:
+            msg = ("The following Fluentd logging configuration problem was found:"
+                   "{}".format(config_error))
+            return {"failed": True, "changed": False, "msg": msg}
+
+        return {"failed": False}
+
+    def check_logging_config(self, task_vars):
+        """Ensure that the configured Docker logging driver matches fluentd settings.
+        This means that, at least for now, if the following condition is met:
+
+            openshift_logging_fluentd_use_journal == True
+
+        then the value of the configured Docker logging driver should be "journald".
+        Otherwise, the value of the Docker logging driver should be "json-file".
+        Returns an error string if the above condition is not met, or None otherwise."""
+        use_journald = get_var(task_vars, "openshift_logging_fluentd_use_journal", default=True)
+
+        # if check is running on a master, retrieve all running pods
+        # and check any pod's container for the env var "USE_JOURNAL"
+        group_names = get_var(task_vars, "group_names")
+        if "masters" in group_names:
+            use_journald = self.check_fluentd_env_var(task_vars)
+
+        docker_info = self.execute_module("docker_info", {}, task_vars)
+        if not docker_info.get("info", False):
+            return "No information was returned by the Docker client. Unable to verify the logging driver in use."
+
+        if not docker_info["info"].get("LoggingDriver", False):
+            return "No logging driver information was returned by the Docker client."
+
+        logging_driver = docker_info["info"]["LoggingDriver"]
```

Comment:
Perhaps it is not terribly useful to distinguish when `docker_info` is empty and when only the `LoggingDriver` is missing?

If so, then we can make the whole block shorter:

```python
docker_info = self.execute_module("docker_info", {}, task_vars)
try:
    logging_driver = docker_info["info"]["LoggingDriver"]
except KeyError:
    return "Unable to determine Docker logging driver."
```